src/preamble/02_what_is_a_module_and_a_package.md

```diff
@@ -1 +1,366 @@
-# What is a module and a package?
+# What is a module, a component, and a package?
+
+## TL;DR
+
+There is an ongoing initiative to standardize the terminology we use to refer to
+the hierarchy of collections of code in the Haskell community. If you are
+interested in reading more about the you can find a more thorough treatment
+[here](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/compiler/units#background).
+
+The most significant concepts in this chapter all refer to ways of dividing and
+aggregating code. They form a hierarchy, and are somewhat defined in terms of
+each other. Here is a list of them in order from smallest to largest are:
+
+1. **Module** - The smallest unit of code defined by GHC and the Haskell
+   language specification.
+2. **Component** - A component is a collection of modules. Usually a `library`,
+   `executable`, or a `test-suite`.
+3. **Unit** - Units are compiled components, while not technically _larger_ than
+   a component, they are further along in the packaging process and therefore
+   closer to a distributable artefact.
```

Comment:
I'm afraid I'm none close to understanding 'unit' now than I was before. Perhaps "Units are components after they've been compiled...", if that's the essence? Are they also somehow registered with a hash they are assigned or something? I guess, not 'packaging process', but 'package building process' (which includes validating and registering?)?

Reply:
So, I don't know if the cabal user guide is a great place to go into deep detail on units. I am in favor of just eliding them altogether. The link does explain them more thoroughly. But I think for the purposes of this guide, any more detail would be confusing. I am interested to hear your thoughts on this.

Reply:
Yes, I think dropping them makes sense (though I may be missing something).